src/vanilla/Templates/ReadmeTemplate.cshtml

```diff
@@ -18,15 +18,15 @@ npm install @(Model.PackageName)
 ### Authentication, client creation and @(Model.GetSampleMethod()?.Name) @(Model.GetSampleMethodGroupName()) as an example.
 @EmptyLine
 ```javascript
-import * as msRest from "ms-rest";
-import { @(Model.Name), @(Model.ServiceModelsName) } from "@(Model.PackageName)";
-const subscriptionId = "<Subscription_Id>";
-const token = "<access_token>";
-const creds = new msRest.TokenCredentials(token);
-const client = new @(Model.Name)(creds, subscriptionId);
-@(Model.GenerateSampleMethod(true)).catch((err) => {
-  console.log('An error ocurred:');
-  console.dir(err, {depth: null, colors: true});
+const msRestAzure = require("ms-rest-azure");
```

Comment:
The README for vanilla will be different from the README for azure. What you have in the README for vanilla is fine. Someone using the generic flavor of autorest.nodejs need not use ms-rest-azure. However for Azure flavor we definitely want the current changes in the PR. I would suggest to add a different template in `src/azure/Templates`

Reply:
I've changed this back for vanilla clients.

Reply:
https://github.com/Azure/azure-sdk-for-node/tree/master/lib/services/graphManagement#authentication-client-creation-and-listing-users-as-an-example
```
var client = new graphRbacManagementClient(credentials, tenantId);
```

Reply:
I'm just going to assume that this comment was supposed to go with my question about tenants and credentials. Is there anything that indicates that an SDK client has in its swagger/CodeModel that would indicate that it requires a tenant?

Reply:
Yup, your assumption is correct. Instead of subscriptionId there is tenantId as a client property.
I think you would want to reflect on the parameters provided to the client constructor. You can see that we are creating the client constructor in the template over [here](https://github.com/Azure/autorest.nodejs/blob/master/src/azure/Templates/AzureServiceClientTemplate.cshtml#L69).

[RequiredConstructorParameters](https://github.com/Azure/autorest.nodejs/blob/master/src/vanilla/Model/CodeModelJs.cs#L253) is a string that gets constructed by iterating over an array. You can abstract that logic into a property named  `Array<Parameter> RequiredConstructorParametersArray` and then use this inside the  current `RequiredConstructorParameters`. The array version can be used to reflect the parameter position and type and can be used to create the client instantiation in the sample

Reply:
In that way you dont have to worry whether it is a subscription based or tenant based client. Tomorrow if someone comes up with a foo based client, it would be fine for us since we would be providing sample values for all the required parameters in the constructor based on their type and position.

Reply:
It seems like there's a decent amount of work to be done to support tenant-based credentials in our generated readme.md sample. I've created https://github.com/Azure/autorest.nodejs/issues/36 to track that work. I'll do it in a future PR.